pyads/ads.py

```diff
@@ -262,6 +263,11 @@ def size_of_structure(structure_def: StructureDef) -> int:
                 num_of_bytes += (str_len + 1) * size
             else:
                 num_of_bytes += (PLC_DEFAULT_STRING_SIZE + 1) * size
+        elif plc_datatype == PLCTYPE_WSTRING:
+            if str_len is not None:
+                num_of_bytes += (str_len + 1) * size
+            else:
+                num_of_bytes += (PLC_DEFAULT_STRING_SIZE + 1) * 2 * size
```

Comment:
For W_STRINGs are these also terminated by a word so should it be + 2? It might be worth adding comments on these now explaining the plus one, and * 2 to explain the "magic" numbers.

Do we also need a * 2  and + 1/2 if the user specifies a length? If you declare a `WSTRING(10)` for example is that 20 + 1/2 bytes?

Reply:
https://infosys.beckhoff.com/english.php?content=../content/1033/tc3_plc_intro/2529437323.html&id=

Reply:
> For W_STRINGs are these also terminated by a word so should it be + 2?
Agreed, should be +2 here. I think I missed that one.

> Do we also need a * 2 and + 1/2 if the user specifies a length? 
The length the user specifies with e.g. WSTING(10) defines the number of characters. For WSTRINGs a character takes two bytes. So the number of bytes needs to be 2* of this value. In the link you sent it says a WSTRING is terminated by a 0 but doesn't tell 1 or 2 Bytes, but by testing with my PLC I found out it is actually two Bytes.

Reply:
Fixed it in #f9f7ffe

Reply:
If the null terminator is 2 bytes, should lines 268,  332, 421, 425, then in the tests lines 1501, 1512,  be `+ 2`, not `+ 1` ? 

Being picky, I'm not sure the `# if no str_len is given use default size` comments are required, I think this is fairly obvious from the `str_len = PLC_DEFAULT_STRING_SIZE`. The comments about the number of character explanations though are useful.

Other than that the code seems to fit with what it did previously.

Reply:
> If the null terminator is 2 bytes, should lines 268, 332, 421, 425, then in the tests lines 1501, 1512, be + 2, not + 1 ?

All lines apply the schema `2 * (size + 1)`  so we have two Bytes null-terminator, the same in the tests. But as I applied the changes just yesterday in https://github.com/stlehmann/pyads/pull/293/commits/f9f7ffe054166b8dcd0d2affc15158e19dcc0c22 and https://github.com/stlehmann/pyads/pull/293/commits/6c606c38e92c38d2c9c2325d00ad7d867228a5f9 you might have missed them.

Reply:
> Being picky, I'm not sure the # if no str_len is given use default size comments are required, I think this is fairly obvious from the str_len = PLC_DEFAULT_STRING_SIZE.

It' s a bit picky 😋 But you are basically right. I can't argue with that so I removed the comment.

Reply:
> > If the null terminator is 2 bytes, should lines 268, 332, 421, 425, then in the tests lines 1501, 1512, be + 2, not + 1 ?
> 
> All lines apply the schema `2 * (size + 1)`  so we have two Bytes null-terminator, the same in the tests. But as I applied the changes just yesterday in https://github.com/stlehmann/pyads/pull/293/commits/f9f7ffe054166b8dcd0d2affc15158e19dcc0c22 and https://github.com/stlehmann/pyads/pull/293/commits/6c606c38e92c38d2c9c2325d00ad7d867228a5f9 you might have missed them.

I think I may have also been slightly stupid as `2 * (str_len + 1) ` is the same thing as ` (2 * str_len) + 2`...😬